app/controllers/users_controller.rb

```diff
@@ -12,7 +12,8 @@ def create
     if @user.save
       render status: :created, action: :show
     else
-      render status: :bad_request, json: @user.errors.add(:email, @user.errors.delete(:unconfirmed_email))
+      @user.errors.add(:email, @user.errors.delete(:unconfirmed_email)) if @user.errors.include?(:unconfirmed_email)
```

Comment:
Can you add tests for the SHOW and UPDATE routes to verify an unconfirmed account produces a status bad request?